specs/operator/auto-prune.spec

```diff
@@ -0,0 +1,405 @@
+      |2   |testdata/pruner/pipeline/pipelinerun-for-pruner.yaml|
+      |3   |testdata/pruner/task/task-for-pruner.yaml           |
+      |4   |testdata/pruner/task/taskrun-for-pruner.yaml        |
+  * Sleep for "120" seconds
+  * Create
+      |S.NO|resource_dir                                        |
+      |----|----------------------------------------------------|
+      |1   |testdata/pruner/pipeline/pipelinerun-for-pruner.yaml|
+      |2   |testdata/pruner/task/taskrun-for-pruner.yaml        |
+  * Annotate namespace with "operator.tekton.dev/prune.keep-since=2"
+  * Update pruner config "without" keep "" schedule "*/1 * * * *" resouces "pipelinerun,taskrun" and "with" keep-since "10"
+  * Sleep for "80" seconds
+  * Assert if cronjob with prefix "tekton-resource-pruner" is "present" in target namespace
+  * "5" number of pipelineruns should be present
+  * "10" number of taskruns should be present
+  * Remove annotation "operator.tekton.dev/prune.keep-since" from namespace
+  * Remove auto pruner configuration from config CR
+  * Assert if cronjob with prefix "tekton-resource-pruner" is "not present" in target namespace
+
+
+## Verify auto prune with keep per namespace with global stratergy keep-since: PIPELINES-12-TC11
+Tags: e2e, integration, auto-prune, admin
+Component: Operator
+Level: Integration
+Type: Functional
+Importance: Critical
+
+This scenario tests auto prune functionality for a namespace with annotation operator.tekton.dev/prune.keep-since and the global stratergy keep-since.
+If the globaly strategy and the strategy of namespace is different, the operator.tekton.dev/prune.strategy=strategy is must
+
+Steps:
+  * Verify ServiceAccount "pipeline" exist
+  * Remove auto pruner configuration from config CR
+  * Create
+      |S.NO|resource_dir                                        |
+      |----|----------------------------------------------------|
+      |1   |testdata/pruner/pipeline/pipeline-for-pruner.yaml   |
+      |2   |testdata/pruner/pipeline/pipelinerun-for-pruner.yaml|
+      |3   |testdata/pruner/task/task-for-pruner.yaml           |
+      |4   |testdata/pruner/task/taskrun-for-pruner.yaml        |
+  * Annotate namespace with "operator.tekton.dev/prune.keep=2"
+  * Update pruner config "without" keep "" schedule "*/1 * * * *" resouces "pipelinerun,taskrun" and "with" keep-since "10"
+  * Sleep for "80" seconds
+  * Assert if cronjob with prefix "tekton-resource-pruner" is "present" in target namespace
+  * "5" number of pipelineruns should be present
+  * "10" number of taskruns should be present
+  * Annotate namespace with "operator.tekton.dev/prune.strategy=keep"
+  * Sleep for "80" seconds
+  * "2" number of pipelineruns should be present
+  * "2" number of taskruns should be present
+  * Remove auto pruner configuration from config CR
+  * Assert if cronjob with prefix "tekton-resource-pruner" is "not present" in target namespace
+
+## Verify auto prune shcedule per namespace: PIPELINES-12-TC12
```

Comment:
This one failed on my cluster. I fear that `Sleep for "xx" seconds` might cause instability in these tests. We don't need to fix in immediately in this PR, though.